agent/tcs/client/client.go

```diff
@@ -345,3 +359,100 @@ func copyTaskHealthMetrics(from []*ecstcs.TaskHealth) []*ecstcs.TaskHealth {
 	copy(to, from)
 	return to
 }
+
+// publishInstanceStatus queries the doctor.Doctor instance contained within cs,
+// converts the healthcheck results to an InstanceStatusRequest and then sends it
+// to the backend
+func (cs *clientServer) publishInstanceStatus() {
+	// Note to disambiguate between health metrics and instance statuses
+	//
+	// Instance status checks are performed by the Doctor class in the doctor module
+	// but the code calls them Healthchecks. They are named such because they denote
+	// that the container runtime (Docker) is healthy and communicating with the
+	// ECS Agent. Container instance statuses, which this function handles,
+	// pertain to the status of this container instance.
+	//
+	// Health metrics are specific to the tasks that are running on this particular
+	// container instance. Health metrics, which the publishHealthMetrics function
+	// handles, pertain to the health of the tasks that are running on this
+	// container instance.
+	if cs.pullInstanceStatusTicker == nil {
+		seelog.Debug("Skipping publishing container instance statuses. Publish ticker is uninitialized")
+		return
+	}
+
+	for {
+		select {
+		case <-cs.pullInstanceStatusTicker.C:
+			if !cs.doctor.HasStatusBeenReported() {
+				err := cs.publishInstanceStatusOnce()
+				if err != nil {
+					seelog.Warnf("Unable to publish instance status: %v", err)
+				} else {
+					cs.doctor.SetStatusReported(true)
+				}
+			} else {
+				seelog.Debug("Skipping publishing container instance status message that was already sent")
+			}
+		case <-cs.ctx.Done():
+			return
+		}
+	}
+}
+
+// publishInstanceStatusOnce gets called on a ticker to pull instance status
+// from the doctor instance contained within cs and sned that information to
+// the backend
+func (cs *clientServer) publishInstanceStatusOnce() error {
+	// Get the list of health request to send to backend.
+	request, err := cs.getPublishInstanceStatusRequest()
+	if err != nil {
+		return err
+	}
+
+	// Make the publish instance status request to the backend.
+	err = cs.MakeRequest(request)
+	if err != nil {
+		return err
+	}
+
```

Comment:
Can we move `cs.doctor.SetStatuReported` to this line from line 392. In my opinion it will be more cleaner as it always needs to be updated when publishing the status which is done by this function.

Reply:
Oh yeah, sure.